### PR TITLE
[Tech Debt] Refactor API user serialization methods

### LINKED ISSUE #2206: [Tech Debt] Refactor API user serialization methods
The API's user serialization and deserialization methods were plugins for the Passport session library prior to #2158. After #2158, the serialization methods are only used by our own code, so they doesn't have to conform to Passport's APIs anymore. It might make sense to refactor them into promise-based functions to match with the rest of our custom code.

### This task is done when...
 - [ ] the `serializeUser` and `deserializeUser` methods in `api/auth/serialization.js` return a promise instead of requiring a callback

### PR DESCRIPTION
**Pull request by pphillips-fearless**
Refactor API user serialization methods

### This pull request changes...

- The API's user serialization and deserialization methods were refactored into promise-based functions
- closes #2206 

### This pull request is ready to merge when...

- [ ] Tests have been updated (and all tests are passing)
- [ ] This code has been reviewed by someone other than the original author
~~- [ ] The experience passes a basic manual accessibility audit (keyboard nav, screenreader, text scaling) OR an exemption is documented~~
~~- [ ] The change has been documented~~
~~- [ ] Associated OpenAPI documentation has been updated~~
- [x] Changelog is updated as appropriate

### CODE REVIEW

File: api/auth/serialization.test.js [hunk 1]
```diff
@@ -21,16 +19,14 @@ tap.test('passport serialization', async serializationTest => {
...
-    await serialization.serializeUser(user, doneCallback, { sessionStore });
-
-    serializeTest.ok(
-      doneCallback.calledWith(null, 'session-id'),
-      'serializes the user object'
-    );
+    serialization.serializeUser(user, { sessionStore }).then(sessionId => {
+      serializeTest.same(sessionId, 'session-id', 'serializes the user object');
+      serializeTest.end();
+    });
```
Comment (mgwalker) [inline api/auth/serialization.test.js:29]: I'm curious about why this was changed from an `await`-type promise to using a `then`, just as a matter of consistency.

File: api/auth/index.test.js [hunk 2]
```diff
@@ -143,7 +143,7 @@ tap.test('authentication setup', async authTest => {
       im: 'weasel',
       ir: 'baboon'
     };
-    serializeUser.yields(null, 'unique-session-id');
+    serializeUser.resolves('unique-session-id');
```
Comment (mgwalker) [inline api/auth/index.test.js:146]: Curious if you think we should add a tech debt issue to add a test for the case where `serializeUser` throws an error?

File: api/auth/serialization.test.js (hunk 1 above)
Comment (pphillips-fearless) [reply api/auth/serialization.test.js:29]: I was on the fence about this but I did it first and was favoring the `then` type.  The second one needed the `await` syntax so you're right we should make them consistent.

File: api/auth/serialization.js [hunk 3]
```diff
@@ -9,45 +9,38 @@ const { addUserSession, getUserIDFromSession } = require('./sessionStore');
...
-        done(null, null);
       }
     } else {
       logger.warn(`${sessionID} does not map to a user ID`);
-      done(null, null);
     }
   } catch (e) {
     logger.error(null, e);
-    done('Could not deserialize user');
+    logger.silly('Could not deserialize user');
```
[the remark this answers is not in the document]
Comment (pphillips-fearless) [reply api/auth/serialization.js:43]: we can do a catch-log-throw here

**Pull request merged**